maven-plugin/src/main/maven/plugin.xml

```diff
@@ -79,6 +86,90 @@
+          <description>inject the project</description>
+        </parameter>
+        <parameter>
+          <name>language</name>
+          <type>string</type>
+          <required>false</required>
+          <editable>true</editable>
+          <description>Generate java or scala code</description>
+        </parameter>
+        <parameter>
+          <name>generateClient</name>
+          <type>boolean</type>
+          <required>false</required>
+          <editable>true</editable>
+          <description>Whether to generate client code</description>
+        </parameter>
+        <parameter>
+          <name>generateServer</name>
+          <type>boolean</type>
+          <required>false</required>
+          <editable>true</editable>
+          <description>Whether to generate server code</description>
+        </parameter>
+        <parameter>
+          <name>generatorSettings</name>
+          <type>java.util.List</type>
+          <required>false</required>
+          <editable>true</editable>
+          <description>extra settings to pass to the generators</description>
+        </parameter>
+        <parameter>
+          <name>extraGenerators</name>
+          <type>java.util.List</type>
+          <required>false</required>
+          <editable>true</editable>
+          <description>extra generators to load</description>
+        </parameter>
+        <parameter>
+          <name>protoPaths</name>
+          <type>list</type>
+          <required>false</required>
+          <editable>true</editable>
+          <description>Where to look for proto files</description>
+        </parameter>
+        <parameter>
+          <name>outputDirectory</name>
+          <type>string</type>
+          <required>false</required>
+          <editable>true</editable>
+          <description>Where to write the generated files</description>
+        </parameter>
+        <parameter>
+          <name>protocVersion</name>
+          <type>string</type>
+          <required>false</required>
+          <editable>true</editable>
+          <description>The proto compiler version</description>
+        </parameter>
+      </parameters>
+      <configuration>
+        <project implementation="" default-value="${project}"/>
+        <language implementation="enum" default-value="JAVA">${akka-grpc.language}</language>
+        <generateClient implementation="boolean" default-value="true">${akka-grpc.generate-client}</generateClient>
+        <generateServer implementation="boolean" default-value="true">${akka-grpc.generate-server}</generateServer>
+        <generatorSettings implementation="java.util.List" default-value=""/>
+        <extraGenerators implementation="java.util.List" default-value=""/>
+        <protoPaths default-value="src/test/proto,src/test/protobuf">${akka-grpc.protoPaths}</protoPaths>
+        <outputDirectory default-value="target/generated-test-sources">${akka-grpc.outputDirectory}</outputDirectory>
```

Comment:
Hmm, that's quite some duplication.

Is this the way other code-generating plugins do it as well?

Reply:
If you develop using maven, the `plugin.xml` is generated based on annotations in the code so it's not a problem that normally comes up as you don't need to maintain it.

I could factor the common XML out and use a templating tool but I'm not that familiar with SBT so I'd appreciate some help on how to do it.

Reply:
Aah makes sense. Adding templating would be possible for example with sbt-twirl, but to be honest I'm not sure the added complexity is worth it. Let's live with the duplication for now ;)